### PR TITLE
Fixes for RenderAtIdealSize

### PR DESCRIPTION
**Pull request by msDestiny14**
I noticed that the offset seemed to be calculated twice so I removed one of them. 

https://github.com/BabylonJS/Babylon.js/issues/10041

### CODE REVIEW

**Comment by msDestiny14** [comment]
If anyone knows of additional examples I should test with to make sure I'm not breaking backwards compatibility please let me know.

File: gui/src/2D/advancedDynamicTexture.ts [hunk 1]
```diff
@@ -501,15 +501,6 @@ export class AdvancedDynamicTexture extends DynamicTexture {
         var renderWidth = engine.getRenderWidth() * this._renderScale;
         var renderHeight = engine.getRenderHeight() * this._renderScale;
 
-        if (this._renderAtIdealSize) {
```
Comment (deltakosh) [inline gui/src/2D/advancedDynamicTexture.ts:504]: Can you elaborate more on why you remove that code?

**Comment by deltakosh** [comment]
You have plenty here: https://doc.babylonjs.com/playground?q=renderAtIdealSize&type=code

File: gui/src/2D/valueAndUnit.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ export class ValueAndUnit {
      * @returns the value
      */
     public getValue(host: AdvancedDynamicTexture): number {
-        if (host && !this.ignoreAdaptiveScaling && this.unit !== ValueAndUnit.UNITMODE_PERCENTAGE) {
+        if (host && !this.ignoreAdaptiveScaling && this.unit !== ValueAndUnit.UNITMODE_PERCENTAGE && host.renderAtIdealSize) {
```
Comment (deltakosh) [inline gui/src/2D/valueAndUnit.ts:78]: This seems incorrect as ideadlWith and height can still be used: when renderAtIdealSize = false;
https://doc.babylonjs.com/divingDeeper/gui/gui#adaptive-scaling

renderAtIdealSize is here to force the texture to rescale always at the expected ideal size

File: gui/src/2D/advancedDynamicTexture.ts (hunk 1 above)
Comment (msDestiny14) [reply gui/src/2D/advancedDynamicTexture.ts:504]: This line was changing the render w/h but then it appearing to be re-accounted for other file. Calculating it twice?

Comment (deltakosh) [reply gui/src/2D/advancedDynamicTexture.ts:504]: but this needs to happen as the goal of renderAtIdealSize is precisely to do that scale (check the doc link I shared)

**Comment by msDestiny14** [comment]
@deltakosh here is the new version :) @RaananW recommended making a visual test case for this but Having some local build issues at the moment. Prioritizing the fix.

**Review by deltakosh — APPROVED** [review]
Looks good!

**Comment by msDestiny14** [comment]
wait @deltakosh does it look like I'm taking @Popov72 's changes from somewhere too? I pull from master before committing. Bad merge?

**Pull request merged**